app/src/main/java/com/devoxx/android/service/WearService.java

```diff
@@ -0,0 +1,647 @@
+
+    @Bean
+    SlotsDataManager slotsDataManager;
+
+    @Bean
+    SpeakersDataManager speakersDataManager;
+
+    @Bean
+    NotificationsManager notificationsManager;
+
+
+    protected GoogleApiClient mApiClient;
+
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        // Connect to Play Services
+        mApiClient = new GoogleApiClient.Builder(this)
+                .addApi(Wearable.API)
+                .build();
+        mApiClient.connect();
+    }
+
+    @Override
+    public void onDestroy() {
+        if ((mApiClient != null) && (mApiClient.isConnected())) {
+            mApiClient.disconnect();
+        }
+
+        super.onDestroy();
+    }
+
+    @Override
+    public void onMessageReceived(MessageEvent messageEvent) {
+
+        // Processing the incoming message
+        String path = messageEvent.getPath();
+        String data = new String(messageEvent.getData());
+
+
+        // send schedules to the Wearable
+        if (path.startsWith(Constants.CHANNEL_ID + Constants.SCHEDULES_PATH)) {
+            sendSchedules();
+            return;
+        }
+
+        // send slots to the Wearable
+        if (path.startsWith(Constants.CHANNEL_ID + Constants.SLOTS_PATH)) {
+
+            try {
+                sendSlots(Long.parseLong(data));
+
+            } catch (Exception ex) {
+                Log.e(TAG, ex.getLocalizedMessage());
+            }
+
+            return;
+        }
+
+        // send the talk to the Wearable
+        if (path.startsWith(Constants.CHANNEL_ID + Constants.TALK_PATH)) {
+            sendTalk(data);
+            return;
+        }
+
+
+        // send the speaker to the Wearable
+        if (path.startsWith(Constants.CHANNEL_ID + Constants.SPEAKER_PATH)) {
+            sendSpeaker(data);
+            return;
+        }
+
+
+        // send the favorite's status of a talk
+        if (path.startsWith(Constants.CHANNEL_ID + Constants.FAVORITE_PATH)) {
+            sendFavorite(data);
+            return;
+        }
+
+        // Add the favorite's status to a talk
+        if (path.equalsIgnoreCase(Constants.CHANNEL_ID + Constants.ADD_FAVORITE_PATH)) {
+            addFavorite(data);
+            return;
+        }
+
+        // Remove the favorite's status of a talk
+        if (path.equalsIgnoreCase(Constants.CHANNEL_ID + Constants.REMOVE_FAVORITE_PATH)) {
+            removeFavorite(data);
+            return;
+        }
+
+        // Twitter
+        if (path.equalsIgnoreCase(Constants.CHANNEL_ID + Constants.TWITTER_PATH)) {
+            followOnTwitter(data);
+            return;
+        }
+
+    }
+
+
+    private void sendToWearable(final PutDataMapRequest putDataMapRequest) {
+
+        if ((mApiClient != null) && (mApiClient.isConnected())) {
+            Wearable.DataApi.putDataItem(mApiClient, putDataMapRequest.asPutDataRequest());
+            return;
+        }
+
+
+        mApiClient = new GoogleApiClient.Builder(getApplicationContext())
```

Comment:
You have few places with the same code (connect to google api), I would extract class for it, and create method that accept action to be done in onConnected. Something like this:

GoogleApiConnector con = new GoogleApiConnector();
con.connectWearApi(Runnable onConnectedAction);

So you have one place where connection to wear api occurs.